presto-cli/src/main/java/com/facebook/presto/cli/ErrorMessages.java

```diff
@@ -88,7 +90,12 @@ public Builder addTip(Tips tip, Object ...toFormat)
 
             public String build()
             {
-                return builder.toString();
+                if (!session.isQuiet()) {
```

Comment:
I would prefer to use `quiet` in the same manner as `debug` is used. So in case when `quiet` is enabled then I would not add any `Tip` to builder. 

What happens when `--debug --quiet` is used?
Make sure you don't display TIP_INTRO when no `Tip` is added.